groovy-language-server/src/test/java/com/palantir/groovylanguageserver/GroovycWrapperTest.java

```diff
@@ -379,18 +383,589 @@ public void testCompile_error() throws InterruptedException, ExecutionException,
         expectedDiagnostics
                 .add(new DefaultDiagnosticBuilder(
                         "unable to resolve class ExceptionNew1 \n @ line 7, column 18.", DiagnosticSeverity.Error)
-                        .range(Ranges.createRange(7, 18, 7, 73))
-                        .source(test1.getAbsolutePath())
-                        .build());
+                                .range(Ranges.createRange(7, 18, 7, 73))
+                                .source(test1.getAbsolutePath())
+                                .build());
         expectedDiagnostics
                 .add(new DefaultDiagnosticBuilder(
                         "unable to resolve class ExceptionNew222 \n @ line 7, column 18.", DiagnosticSeverity.Error)
-                        .range(Ranges.createRange(7, 18, 7, 75))
-                        .source(test2.getAbsolutePath())
-                        .build());
+                                .range(Ranges.createRange(7, 18, 7, 75))
+                                .source(test2.getAbsolutePath())
+                                .build());
         assertEquals(expectedDiagnostics, actualDiagnostics);
     }
 
+    @Test
+    public void testReferences_edgeCases() throws IOException {
+        File newFolder1 = root.newFolder();
+        // edge cases, intersecting ranges
+        File file = addFileToFolder(newFolder1, "Dog.groovy",
+                "class Dog {\n"
+                + "   Cat friend1;\n"
+                + "   Cat2 friend2;\n"
+                + "   Cat bark(Cat enemy) {\n"
+                + "      println \"Bark! \" + enemy.name\n"
+                + "      return friend1\n"
+                + "   }\n"
+                + "}\n"
+                + "class Cat {\n"
+                + "   public String name = \"Bobby\"\n"
+                + "}\n"
+                + "class Cat2 {\n"
+                + "   InnerCat2 myFriend;\n"
+                + "   class InnerCat2 {\n"
+                + "   }\n"
+                + "}\n");
+        // edge case on one line
+        File enumFile = addFileToFolder(newFolder1, "MyEnum.groovy",
+                "enum MyEnum {ONE,TWO}\n");
+        // edge case on one line
+        File innerClass = addFileToFolder(newFolder1, "AandB.groovy",
+                "public class A {public static class B {}\n"
+                        + "A a\n"
+                        + "B b\n"
+                + "}\n");
+        GroovycWrapper wrapper = GroovycWrapper.of(output.getRoot().toPath(), root.getRoot().toPath());
+        Set<Diagnostic> diagnostics = wrapper.compile();
+        assertEquals(0, diagnostics.size());
+
+        // InnerCat2 references - testing finding more specific symbols that are contained inside another symbol's
+        // range.
+        assertEquals(Sets.newHashSet(
```

Comment:
there's inconsistency between assertEquals and assertThat in this class -- we should fix this later